ports/qtbase/portfile.cmake

```
@@ -150,9 +152,12 @@ else()
     list(APPEND FEATURE_NET_OPTIONS -DINPUT_openssl=no)
 endif()
 
+if (NOT "dnslookup" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_WINDOWS)
+    list(APPEND FEATURE_NET_OPTIONS -DCMAKE_DISABLE_FIND_PACKAGE_WrapResolv:BOOL=ON)
+endif()
```

```suggestion
if ("dnslookup" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_WINDOWS)
    list(APPEND FEATURE_NET_OPTIONS -DCMAKE_REQUIRE_FIND_PACKAGE_WrapResolv:BOOL=ON)
else()
    list(APPEND FEATURE_NET_OPTIONS -DCMAKE_DISABLE_FIND_PACKAGE_WrapResolv:BOOL=ON)
endif()
```

However doing it directly in vcpkg_check_features is probably better

---

I'm happy to apply the change.
But looking at the acutal Qt cmake file i think the whole if/else should be removed.

https://github.com/qt/qtbase/blob/dev/src/network/CMakeLists.txt#L237-L242
```
qt_internal_extend_target(Network CONDITION QT_FEATURE_dnslookup AND QT_FEATURE_libresolv
    SOURCES
        kernel/qdnslookup_unix.cpp
    LIBRARIES
        WrapResolv::WrapResolv
)
```


WrapResolv is only used if dnslookup is enabled.

If dnslookup is disabled, CMAKE_DISABLE_FIND_PACKAGE doesn't change anything as WrapResolv isn't linked either way
If enabled, WrapResolv is required, otherwise there will be runtime errors.

---

vcpkg needs to be explicit with those dependency even if the FEATURES would deactivate the lookup. That is because an update might change the behavior. 

The error will be earlier if CMAKE_REQUIRE_FIND_PACKAGE_WrapResolv is set.

---

I see.

I decided to go with the conditional as i was unsure how to do it in vcpkg_check_features only on !windows.